bootstrap/helpers/services.php

```
@@ -115,65 +115,163 @@ function updateCompose(ServiceApplication|ServiceDatabase $resource)
         }
 
-        $serviceName = str($resource->name)->upper()->replace('-', '_')->replace('.', '_');
-        $resource->service->environment_variables()->where('key', 'LIKE', "SERVICE_FQDN_{$serviceName}%")->delete();
-        $resource->service->environment_variables()->where('key', 'LIKE', "SERVICE_URL_{$serviceName}%")->delete();
+        // Extract SERVICE_URL and SERVICE_FQDN variable names from the compose template
+        // to ensure we use the exact names defined in the template (which may be abbreviated)
+        // IMPORTANT: Only extract variables that are DIRECTLY DECLARED for this service,
+        // not variables that are merely referenced from other services
+        $serviceConfig = data_get($dockerCompose, "services.{$name}");
+        $environment = data_get($serviceConfig, 'environment', []);
+        $templateVariableNames = [];
```

PHPMD (2.15.0)

Avoid excessively long variable names like $templateVariableNames. Keep variable name length under 20. (undefined)

(LongVariable)